### ISSUE TITLE
Fix Request: false positive in HashCodeOnlyCallsSuper

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#440]
```
class EqHashTryoutBad6 {
    String field1;
    String field2;// bad

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EqHashTryoutBad3 that = (EqHashTryoutBad3) o;
        return Objects.equals(field1, that.field1) &&
                Objects.equals((field2.toLowerCase()), that.field2.toLowerCase());
    }

    @Override
    public int hashCode() {
        return Objects.hash(field1, field2); // <-- false positive
    }
}


class EqHashTryoutBad8 {
    String field1;

    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EqHashTryoutBad8 that = (EqHashTryoutBad8) o;
        return Objects.equals(field1, that.field1);
    }
    public int hashCode() {
        return super.hashCode(); // bad
    }
}
```
